gobblin-runtime/src/main/java/org/apache/gobblin/runtime/messaging/hdfs/FileSystemMessageBuffer.java

```diff
@@ -86,39 +106,79 @@ private void scheduleTaskForPollingNewMessages() {
   }
 
   private List<DynamicWorkUnitMessage> getNewMessages() {
-    // STUB: TODO GOBBLIN-1685 Read all files in dir, deserialize, then clean up byte files in dir
-    return null;
+    List<DynamicWorkUnitMessage> messageList = new LinkedList<>();
+    try {
+      FileStatus[] fileStatus = fs.listStatus(path);
+      for(FileStatus status : fileStatus){
+        Path singlePath = status.getPath();
+        if (fs.isFile(singlePath))
+        {
+          FSDataInputStream fis = fs.open(singlePath);
+          DynamicWorkUnitMessage message = DynamicWorkUnitSerde.deserialize(ByteStreams.toByteArray(fis));
+          messageList.add(message);
+          fs.delete(singlePath, true);
+          log.debug("Fetched message from HDFS path {}.", singlePath.toString());
+        }
+      }
+    } catch (IOException ioException) {
+      log.error("Failed to get message from HDFS path {}.", path.toString(), ioException);
+    }
+
+    return messageList;
+  }
+
+  private void persistMessage(byte[] serializedMsg, DynamicWorkUnitMessage message) throws IOException {
```

Comment:
nit: having message passed in twice is a bit strange. Either have it be something like
```
private void persistMessage(DynamicWorkUnitMessage message)

```
or
```
  @Override
  public void publish(DynamicWorkUnitMessage item) throws IOException {
    byte[] serializedMsg = DynamicWorkUnitSerde.serialize(item);
    Path p = this.getPathForMessage(item);
    persistMessage(serializedMsg, p);
  }

...
private void persistMessage(byte[] serializedMsg, Path p) throws IOException {

```

Reply:
I moved whole persistMessage logic into publish method, since it's just a couple lines of code